src/main/java/com/fundynamic/d2tm/game/entities/units/Unit.java

```diff
@@ -491,4 +488,49 @@ public Entity getEntityToAttack() {
     public Vector2D getDimensions() {
         return Vector2D.create(TILE_SIZE, TILE_SIZE);
     }
+
+    public void harvest(Coordinate target) {
+        moveTo(target);
+    }
+
+    private void setState(UnitState state) {
```

Comment:
@arjenvanderende look... I had to add this if statement to make sure things kept working.

I think it is mostly due the limbo state between state machine and non-state machine pattern currently within the unit (as it would 'set state' in every update cycle, ie see `harvesting` call being called in the `update` method).